girder/models/model_base.py

```diff
@@ -281,6 +281,45 @@ def textSearch(self, query, offset=0, limit=0, sort=None, fields=None,
 
         return cursor
 
+    def prefixSearch(self, query, offset=0, limit=0, sort=None, fields=None,
+                     filters=None, prefixSearchFields=None):
+        """
+        Search for documents in this model's collection by a prefix string.
+        The fields that will be searched based on this prefix must be set as
+        the ``prefixSearchFields`` attribute of this model, which must be an
+        iterable. Elements of this iterable must be either a string representing
+        the field name, or a 2-tuple in which the first element is the field
+        name, and the second element is a string representing the regex search
+        options.
+
+        :param query: The prefix string to look for.
+        :type query: str
+        :param filters: Any additional query operators to apply.
+        :type filters: dict
+        :param prefixSearchFields: To override the model's prefixSearchFields
+            attribute for this invocation, pass an alternate iterable.
+        :returns: A pymongo cursor. It is left to the caller to build the
```

Comment:
1. We can document it as simply an "iterable of documents" if it makes more sense. The cursor has some additional functionality on it e.g. `count()` that could be useful, though.
2. Not sure what you mean by this question, could you elaborate?